DEVELOPMENT.md

```diff
@@ -99,6 +100,43 @@ In general most code is expected to work on Android (but might not yet) with the
 * Functionality aimed at the developer environment, such as console and development provider
 * Any JMX related code.
 
+### The ASF / Github Integration
+
+The Quarks code is in ASF resident git repositories.  
+The repositories are mirrored on github:
+
+    https://github.com/apache/incubator-quarks
+
+Use of the normal github workflow brings benefits to the team including 
+lightweight code reviewing, automatic regression tests, etc 
+for both committers and non-committers.  
+
+For a description of the github workflow see:
+
+    https://guides.github.com/activities/hello-world/
+
+In summary:
+    * fork the incubator-quarks github repository
+    * clone your fork, use lightweight per-task branches, and commit / push changes to your fork
+      * name your branch quarks-xxxx - typically xxxx is a Jira ticket number
+    * when ready, create a pull request.  Committers will get notified.
+      * include QUARKS-XXXX in the name of your pull request
+      * for early preview / feedback, create a pull request with [WIP] in the title.  
+        Committers won’t consider it for merging until after [WIP] is removed.
+
+Since the github incubator-quarks repository is a mirror of the ASF repository,
+the usual github based merge workflow for committers isn’t supported.
+
+Committers can use one of several ways to ultimately merge the pull request
+into the repo at the ASF. One way is described here:
+
+    http://mail-archives.apache.org/mod_mbox/incubator-quarks-dev/201603.mbox/%3C1633289677.553519.1457733763078.JavaMail.yahoo%40mail.yahoo.com%3E
+
+Notes with the above PR merge directions:
+  * seems like it should be using  `git merge --squash pr-<N>`
```

Comment:
Not sure about that, though it would be good to have a reason why 'squash' should be used.

Not using squash does preserve the history of commits and allow others to cherry-pick individual commits.

My opinion would be to not squash (for the above reasons) and if a contributor doesn't want their history of changes to be recorded they squash before creating a merge request.